nqeit-nessie-common/src/main/java/org/projectnessie/integtests/nessie/internal/AbstractStorageLocation.java

```diff
@@ -82,7 +82,10 @@ public URI getUri() {
 
   @Override
   public void close() throws Throwable {
-    deleteRecursively(tempDir);
+    if (tempDir != null) {
```

Comment:
Not sure why this change is required for this PR.
Sure, being on the safe side is okay, but it seems that this temp-dir resource is added although it's not used?

Reply:
https://github.com/projectnessie/query-engine-integration-tests/blob/80d275a9d8a2ba4832c7a464076de6d514e6c177/nqeit-nessie-common/src/main/java/org/projectnessie/integtests/nessie/internal/AbstractStorageLocation.java#L51-L55

If the warehouse location is null then only we are setting the tempDir. In our case, we are using the warehouse location as S3, so if we do not handle this case, we will get error.

Reply:
In this case, it would be preferable to let `getPath()` throw an `IllegalStateException`, if `path` is null.

Reply:
when location is not null, then `tempDir` is not initialized, but this close method tries to clean it up unconditionally.
i am not sure how it is related to `getPath()´ but even if `getPath()` threw an exception as suggested, then the close method would still be wrong without this change, because it calls `deleteRecursively(null)`, no?

Reply:
Now, getPath() throws an IllegalStateException, if the path is null.